cmd/wallet-web/src/router/index.js

```diff
@@ -71,7 +71,30 @@ router.beforeEach(async (to, from) => {
       console.error('error authenticating user: invalid hash received');
       return false;
     }
-  } else if (to.matched.some((record) => record.meta.requiresAuth)) {
+  }
+
+  if (to.meta.requiresDIDComm) {
+    // Get enableDIDComm from wallet config
+    if (store.getters.getEnableDIDComm === undefined) {
+      if (store.dispatch('loadUser') && store.getters.getEnableDIDComm === undefined) {
+        await store.dispatch('initOpts');
+      }
+    }
+
```

Comment:
Can we simplify this block to something like below?

> if (store.getters.getEnableDIDComm === undefined && store.dispatch('loadUser')) {
>    await store.dispatch('initOpts');
>  }

Reply:
Your suggested change would fetch from local storage every time. We should only do it if `getEnableDIDComm` is `undefined` in Vuex

Reply:
I don't think those are equal statements. `store.dispatch('loadUser')` does not return anything, so in your suggestion we would never dispatch if `store.getters.getEnableDIDComm` is undefined after loading the user from local storage. However, this might be cleaner:
```js
    if (store.getters.getEnableDIDComm === undefined) {
      store.dispatch('loadUser');
      if (store.getters.getEnableDIDComm === undefined) {
        await store.dispatch('initOpts');
      }
    }
```

Reply:
The `loadUser` doesn't define getEnableDIDComm at the moment